### PR TITLE
Support minting American options from PsyOptions program

### PR DESCRIPTION
**Pull request by Taylor123**
Proposal instruction support for minting of American option/writer tokens to DAO or manually input address.

Successful devnet transactions:
- proposal creation which initializes option based on input parameters: https://explorer.solana.com/tx/4JwzQNrpmNzaJTvF477QdXCq1BEfF1gndSFnRVHUKjXPxytBVWW1Z38FX6P2Tm8tNo1FpTCJE5DKM5hYLfa9vvd7?cluster=devnet
- proposal execution which transfers asset to pool and mints option/writer tokens https://explorer.solana.com/tx/3nho4cfFQvoMkzCQWaJpuWaUkjCKhDMKCFKmAL29idRa3jPc8khVbcwEiirx7hDLUEqoGmJ57gAmgfcmh9xUSinT?cluster=devnet

### CODE REVIEW

File: pages/dao/[symbol]/proposal/components/instructions/PsyFinance/MintAmericanOptions.tsx [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+import React, { useContext, useEffect, useReducer } from 'react'
+import Input from '@components/inputs/Input'
+import Tooltip from '@components/Tooltip'
+import useGovernanceAssets from '@hooks/useGovernanceAssets'
+import { AssetAccount } from '@utils/uiTypes/assets'
+import {
+  PsyFinanceMintAmericanOptionsForm,
+  UiInstruction,
+} from '@utils/uiTypes/proposalCreationTypes'
+import { useState } from 'react'
```
Comment (GregoryNEUT) [inline pages/dao/[symbol]/proposal/components/instructions/PsyFinance/MintAmericanOptions.tsx:16]: nit: react import is duplicated

```
import React, { useContext, useEffect, useReducer, useState } from 'react'
```

File: pages/dao/[symbol]/proposal/components/instructions/PsyFinance/MintAmericanOptions.tsx [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+    writerTokenDestinationAccount: '',
+  })
+
+  const shouldBeGoverned = !!(index !== 0 && governance)
+
+  const getInstruction = async (): Promise<UiInstruction> => {
+    if (
+      !underlyingMintInfo ||
+      !form.quoteMint ||
+      !form.strike ||
```
Comment (GregoryNEUT) [inline pages/dao/[symbol]/proposal/components/instructions/PsyFinance/MintAmericanOptions.tsx:77]: Is `0` a valid value for `strike`?

File: utils/uiTypes/proposalCreationTypes.ts [hunk 3]
```diff
@@ -391,6 +392,18 @@ export interface MangoMakeChangeReferralFeeParams2 {
   refMngoRequired2: number
 }
 
+export interface PsyFinanceMintAmericanOptionsForm {
+  contractSize: number
+  expirationUnixTimestamp: number
+  optionTokenDestinationAccount: string
+  quoteMint: string
+  size: number | undefined
+  strike: number | undefined
```
Comment (GregoryNEUT) [inline utils/uiTypes/proposalCreationTypes.ts:401]: In the instruction form, `strike` is set to `0` by default. I believe it never can be `undefined` then

**Review by GregoryNEUT — COMMENTED** [review]
Hello, few comments, hope it helps 🙏

File: pages/dao/[symbol]/proposal/components/instructions/PsyFinance/MintAmericanOptions.tsx (hunk 2 above)
Comment (Taylor123) [reply pages/dao/[symbol]/proposal/components/instructions/PsyFinance/MintAmericanOptions.tsx:77]: It's valid at the protocol level. However, a strike of 0 would be the same as giving the underlying for free so it doesn't make sense at the user level.

File: package.json [hunk 4]
```diff
@@ -52,6 +52,7 @@
     "@mean-dao/msp": "3.0.1-alpha.13",
     "@metaplex-foundation/js": "0.17.5",
     "@metaplex-foundation/mpl-token-metadata": "~2.5.1",
+    "@mithraic-labs/psy-american": "0.2.3",
```
Comment (SebastianBor) [inline package.json:55]: We are trying to minimise the dependencies. Would it be possible to implement this without this dependency?

**Pull request merged**